src/Conversations/Conversation.php

```diff
@@ -356,56 +371,83 @@ public function getId(): ?int
         return $this->id;
     }
 
-    public function setId(int $id)
+    public function setId(int $id): Conversation
```

Comment:
Does a type hint of `self` instead of `Conversation` allow this to be extended more easily?

Reply:
I'm not sure how that would make it easier to extend. Can you provide an example or explain?

Reply:
@davidstanley01 That's my question. I don't know if it makes a difference - I have tended to type hint `self` and then add a phpdoc comment for `@return $this` to indicate that it is a fluent interface instead of an immutable return. I don't know if there is a real difference between `self` and `Conversation` in this context though.

Reply:
I'm going to merge this and we can come back to it later if that works for you...